addons/web/static/src/core/position_hook.js

```diff
@@ -81,7 +82,14 @@ const DEFAULTS = {
  */
 function getBestPosition(reference, popper, { container, margin, position }) {
     // Retrieve directions and variants
-    const [directionKey, variantKey = "middle"] = position.split("-");
+    let [directionKey, variantKey = "middle"] = position.split("-");
+    if (localization.direction === "rtl") {
+        if (variantKey === "start") {
+            variantKey = "end";
+        } else if (variantKey === "end") {
+            variantKey = "start";
+        }
+    }
```

Comment:
Do we really want to switch between right-end and right-start and also between left-start and left-end when we are in RTL ?
I don't think so.

IMHO, we should not make the position hook dependent of the localization.

The strategy used in the Dropdown component is to pass the proper positioning options depending on the l10n direction. I think this is what we should do when we face a similar use case.